### ISSUE TITLE
Requested Trains and rollingstock

### ISSUE THREAD

**Issue by EternalBlueFlame** [EternalBlueFlame/Traincraft-5#58]
This issue thread is for train and rollingstock idea/model submissions.

Notice: you must sign the CLA, found in the README.md of the Project if you submit a model and/or texture

**We are accepting new train and rollingstock textures now!**
For more information, check out my tutorial on adding them in!
see: [https://github.com/EternalBlueFlame/Traincraft-5/commit/b5831c68f4d859a765d86d8d27219aad44715a7c](https://github.com/EternalBlueFlame/Traincraft-5/commit/b5831c68f4d859a765d86d8d27219aad44715a7c)
It was also brought to my attention that if the render enum doesn't have the _ it messes with our naming conventions, so that has to be done as well, here is an example of that:
[https://github.com/EternalBlueFlame/Traincraft-5/commit/b2e3e8200354dd77af3d6edd2538c757929860af](https://github.com/EternalBlueFlame/Traincraft-5/commit/b2e3e8200354dd77af3d6edd2538c757929860af)

Supported Texture Formats:
1: Portable Network Graphics (.png)

**The following is to be dealt with only after the errors have been fixed and the code cleaned up.**

Always remember;
- It's good to share some pictures from your submitted ideas
- Organize the suggestion using Markdown for your posts.
- **Do not** submit or request something that is potentially copyrighted material, these will be denied immediately (Thomas the tank engine, Chuggington, The Polar Express, Big One, just as some examples).
- If your rollingstock has special features, please elaborate on them in detail using a spoiler.
- If you can include the stats of the train like top speed, acceleration, length, height, width, etc that helps us a lot.

If you can submit an art asset that matches Traincraft's style in .java format formats, optionally with textures, your request will be given a priority.

**NOTICE:** Wavefront (.OBJ) is NOT and will NOT be supported for anything other than TileEntites (Rails for example)

--Thanks to @NitroxydeX for the help on the rules

**Comment by bigFISH496** [comment]
I have no experience with modeling in mc, so if someone can do something with these, that'd be awesome 

Alright, I've been sitting on these ideas for a while now, so here goes:
Locomotives:
- 2-8-4 Berkshire class steam locomotive, probably similar in size to the JNR (this is the same style of locomotive seen in "The Polar Express")

//ETERNAL EDIT - dude, copyrights, for legality reasons we're not even supposed to mention it's similar if it is to be used.
- 4-4-0 American style steam locomotive
- 0-6-0 steam switcher
- Any members of the EMD E or F class diesel locomotives
- EMD F40ph diesel locomotive (a staple of modern passenger travel)
- P42 diesel locomotive
- GE GEVO series of diesel locomotives
- More paint schemes for existing locomotives, ie; class II and III railroads, and fallen flags/heritage schemes like Pennsylvania, Nickel Plate Road, CNW, Burlington, Western Pacific, etc.
- More variety of whistles

Wagons:
More passenger cars! Especially more modern ones

//ETERNAL EDIT: be specific.

**Comment by NitroxydeX** [comment]
To everyone that is writing in this thread:
Keep in mind that you are writing with/to developers and graphic artists. We really like formatted texts. Some basic rules for your posts:

- Google is your friend, share us some pictures from your submitted train ideas
- Posts with finished models are 100% more welcome :) - we can use google ourselves to get ideas for trains
- Use Markdown for your posts.
- Sign the CLA of the Project if you submit a train-model/texture

**Comment by LoneSniperSG** [comment]
bigFISH, TrainCraft already contains an American 4-4-0 steam engine, as well as two other locomotives of the same era (4-4-4 Forney & 2-6-0 Mogul). Traincraft also already contains a German-design 0-6-0 steam switcher locomotive, although he is underpowered in my experience.

TrainCraft is the whole reason I started playing MineCraft at all. I love the mod and I would hate to see it fall on its face. I think we could use one or two locomotives ( I like the Berkshire idea), but I can't really think of any holes in the rolling stock. All the bases are pretty much covered there. I would like a ballast undercutter car like RailCraft has and a different variety of passenger car such as American streamliner cars (especially of a 1950's E- or F-type diesel is considered) or modern-style Amtrak cars.
One other piece I have thought of is a flat car that DOESN'T carry people (seriously, why?). How about a flat car that just carries blocks and displays them on its model?

Overall I think the train 'tech tree' and the locomotive power values should be refined more so than adding in new trains. Case in point: my favored SD40 diesel vs the SD70 diesel. Both have the exact same "MineCraft Horsepower", despite the SD70 being 20 years more modern than the SD40. While the comparison is somewhat true in the real world, it makes for little reason to bother crafting SD70's in-game. As I've come to notice especially in the last year, most of the rolling stock issues are those of programming. Playing on 1.6.4 right now, the GP40 has gone missing for some reason. I would like to see his return too.

Happy to elaborate on anything if needed.

**Comment by bigFISH496** [comment]
I didn't realize there was a 4-4-0, my bad. And sorry blue, I should have clarified, the Polar Express was based off of a real Berkshire, Pere Marquette 1225, which still runs excursions in Michigan

**Comment by Valaktik** [comment]
Heyllo ! I am modeling a high speed train for TC ^^ And I need your opinion ! [Image](http://i.imgur.com/BR4i5na.png)
This is the Atlantic TGV in France ! [Image](https://www.sterlingot.com/TGV/tgva-lbes1.jpg)
Ok, that's just a little start but please can you  tell me if it has good proportions or if you see wrong things ... 
I have studies so this train is not to be finished early ... :)

**Comment by EternalBlueFlame** [comment]
There are plenty of example models here that you can take a look at to be sure the quality, positioning, and scale are correct, that should help.
https://github.com/EternalBlueFlame/Traincraft-5/tree/1.7/src/main/java/train/client/render/models

**Comment by Valaktik** [comment]
Oh yes thank you that will help me a lot !

**Comment by melindaendv** [comment]
Just some ideas

Locomotives:

EMD F7 A and B units - Liveries - Santa Fe, Union Pacific, Burlington Northern, Pennsylvania, Kansas City Southern, Southern Pacific. - http://www.american-rails.com/images/KSCUS3405.jpg

GE U30C - Union Pacific, Conrail, Burlington Northern, Seaboard System, BC Rail - http://cdn3.volusion.com/ztna9.tft5b/v/vspfiles/photos/ATL-7343-2.jpg?1456472651

GE Dash 9 - Union Pacific, BNSF, Montana Raillink, Southern Pacific, CN, KCS - http://vignette2.wikia.nocookie.net/trains-and-locomotives/images/5/5c/ATSF_C44-9CW.jpg/revision/latest?cb=20131215235802

P42DC - Amtrak, Via Rail - http://img13.deviantart.net/6c7a/i/2005/331/2/4/amtrak_ge_p42dc_58_by_seangulden.jpg

F59PH - Go Transit, TRE, Metra, Cal Train - http://trainiax.net/drawings/55-paint/tre/55-tre-568-emd-f59ph.GIF

F40PH - Amtrak, Metra, Cal train, Montana Raillink, Burlington Northern - http://www.blwnscale.com/blwphotos/ka-f40ph-amtrak.jpg

GP38-2 - Union Pacific, CN, Montana Raillink, BNSF, Burlington Northern, KCS - http://donsdepot.donrossgroup.net/dr0404/kcs4019.jpg

Rolling Stock:

Bombardier BiLevel coaches with liveries for - Go Transit, Caltrain, TRE, Metra -
http://www.wig-wag-trains.com/Athearn-Pages/Athearn%20Pics/Passenger/CommuterCars/24449_TRE.jpg

Am Fleet coaches - Amtrak  - http://donsdepot.donrossgroup.net/dr0702/at44218.jpg

BiLevel Coaches - Amtrak. Metra, Cal Train, Via Rail - https://katousa.com/HO/Bilevel/35-6036.jpg

Covered hopper or ACF Centerflow car - http://www.progressiverailroading.com/railproducts/graphics/CACH-ARI-PRESS16.jpg

Spine car - http://www.bowser-trains.com/images/Freight/Spine/40541.jpg

EOTD - http://4.bp.blogspot.com/-ZMG9xZs40AQ/UK0otyb9QfI/AAAAAAAALb8/K0_0R2Y_wBA/s1600/EOT-Device.jpg

**Comment by Lanthan64** [comment]
Hi Everyone,

I'd like to describe why I think the TRAXX 2 class might be a great addition to Traincraft.

First of, as requested, here are some links to the manufacturers site where You can find technical data and drawings.
Electric variant: 
http://www.bombardier.com/en/transportation/projects/project.TRAXX-F140-MS-Re484-SBB-Cargo-Switzerland.html
Diesel variant: 
http://www.bombardier.com/en/transportation/projects/project.TRAXX-P160-DE-ME-DB-Regio-Germany.html

The TRAXX-Units are modern electric / diesel locomotives that are used by railways across the world,
mainly Europe, but also Israel and South Africa. Derived classes are as well used in North America.
- So I am talking about a class of locomotives that will be known to a lot of players and not just a few locals.
- Also its modern design includes large front windows that allow viewing the line ahead of the train without switching to an external camera perspective.

There are multiple variants that are visually almost identical, this applies to both electric and diesel versions.
- Which means You can get two locomotives out of a single model. The "pantographs" of the diesel variant just need to be painted transparent.
- The player can freely choose between the two versions depending on what fuel he / she considers easier to acquire. 

I know adding trains is not Your highest priority right now, but once it gains relevancy I'd be happy if these units were along them.

**Comment by melindaendv** [comment]
@cuberail - I work for a railroad in the US and I have never heard or seen that locomotive until now... Interesting as it is there are no known class A railroads (AmTrak, Union Pacific, BNSF, KCS, NS, CSX) in the US using this locomotive either, Maybe limited experimental use on a commuter service like Metra, TRE, or CalTrain but nothing in the freight hauling or overland passenger service.
The only thing I found was Go-Transit in Canada that has a cab car variant (very very few) of what you posted:
https://i.ytimg.com/vi/PT6uu5eYELQ/maxresdefault.jpg

Not saying it would not be an interesting addition to the mod, any train would be a good addition to the mod just in the US there are none of these running regular services..

**Comment by Valaktik** [comment]
@melindaendv AmTrak is using that locomotive :) See : [Image](http://www.railcolor.net/images/basic/siem_216010_56.jpg)
(May be it's not exaclty the same)

**Comment by dymondy2k** [comment]
If I could add my two cents I would love to see a EMD SW-7 Switcher and the standby for every model railroad.. the F7A and B.

**Comment by EternalBlueFlame** [comment]
Good news! We are now accepting textures for existing trains and rollingstock, check the top of this thread for more info.

**Comment by Pro-shadows** [comment]
I am thinking of making some models for new trains, just wondering which program you use to model the trains and rolling stock in? i dont want any cross comparability issues

**Comment by NitroxydeX** [comment]
@Pro-shadows MrBrutal has a tutorial for this: http://blog.meansoft.si/2016/07/31/3d-modeling-for-minecraft-1-software-part-1/

**Comment by EternalBlueFlame** [comment]
@Pro-shadows side from the link nitro provided you should also be able to use older versions of MCAnimator (as long as they support .java output) and if you plan to _really_ get into doing graphics for minecraft, and you have some extra money, I'd suggest looking into BDCraft Cubik.

**Comment by Pro-shadows** [comment]
the models are coming along nicely. now i just need to code

**Comment by ArgonTheConqueror** [comment]
Might I suggest a locomotive?
Vietnam Railways' D19E: Diesel locomotive, Co'Co' wheel arrangement.
![d19e-921](https://cloud.githubusercontent.com/assets/24211198/20651188/3cd8bfc6-b4ae-11e6-86e6-bc5012eb64e7.jpg)
For the power of the locomotive, I think 500-600 MHP would be great, along with a max speed of 90-120 km/h. You decide.
height: 3.9m, width, 2.9m, length, 16.9m.
Acceleration is around the same speed as the ChME3 or the other diesel freight locomotives. Same with braking capability.
Anyway, thanks for reading this, if you are, and sorry to bother you with a request.
Note: The locomotive is a double cab locomotive.
For any other pieces of information, you can go to this link to see. It has images. http://www.railwaysinvietnam.com/D19E.html
If you do intend to put the locomotive in to Traincraft, I have a mp3 file of the train horn if that helps.

**Comment by EternalBlueFlame** [comment]
Can I just congratulate you @ArgonTheConqueror ? This is by far the most well put-together request we have received and it will really help save us effort on having to research that information when we are ready to start adding new trains and have an artist to design them.

**Comment by ArgonTheConqueror** [comment]
For the top view of this locomotive, I found this video. I'll try to find others that have a better resolution. http://www.dailymotion.com/video/x4jk6us 
Sorry that there are no pictures of the locomotive from the top, I'm still looking for better ones.

Note: If you do intend to create this locomotive, it will be a lot wider than the rest, assuming that you are making it proportionately to the size of the Traincraft rail, since the locomotive is metre gauge. It'll be at least 1/4 a block wider than the rest of the locomotives.
I think the image below would help you guys in determining the size. 
![535p9313ax](https://cloud.githubusercontent.com/assets/24211198/20861511/058ea01a-b960-11e6-81d9-0f92f14146a9.jpg)
Side view: 
![side](https://cloud.githubusercontent.com/assets/24211198/21117859/6febc8e0-c089-11e6-9be4-4c4d01758637.JPG)

**Comment by Valaktik** [comment]
Which type of 3D models formats can you accept ? I try making a wagon with Blender and it can export models in Collada (.dae) files, 3D Studio (.3ds), FBX (.fbx), stanford (.ply), [wavefront (.obj) not accepted as said at the top], X3D Extensible 3D (.x3d) and finaly Stl (.stl).

**Comment by NitroxydeX** [comment]
@Valaktik we only support JAVA models. Join our Discord and ask MrBrutal for help. He can provide you a script to convert obj to java.

**Comment by EternalBlueFlame** [comment]
@Valaktik Technie is the main program to use for the port, but you can also make the models using SMP toolbox (which is easier and has more features) if you are willing to wait till the rewrite.
I stumbled across some code for obj today that may let me put that format as trains in the rewrite, but i need to experiment with it more to see if it will really work, because the geometry names are necessary for animation and I'm not fully sure if i can make it read those.

**Comment by Valaktik** [comment]
Ok ! I'll deal with it ... @NitroxydeX are you sure it's possible ?

I don't use techne because it don't have enough feature and it has a bug, 360° is not 360° xD The rotation tool is badly done ok it's not so serious but sometimes the left part of a vehicule is not like the right part :S

Finaly i'll use smp Toolbox it's really cool !

**Comment by NitroxydeX** [comment]
@Valaktik really sure. _"With other software like Cinema, it’s a bit more complicated. You have to have the right scale when exporting, and the file has to be run through a python script afterward. But don’t worry, I’ll provide said script when we get to that part."_ -> http://blog.meansoft.si/2016/07/24/so-you-want-to-make-3d-models-for-minecraft/ under 3. Exporting

**Comment by EternalBlueFlame** [comment]
Turbo Model Thingy (AKA TMT, the render in the rewrite and flans mod) has a programmatic way of converting obj models to java models. it's real-time so you can't get the converted file, but the process still happens and its working proof that the types can be converted between.

My issue is I'm still trying to get it to read model tags so various parts can be moved/animated/rotated/etc.
That's the only thing keeping me from obj trains in the rewrite.

**Comment by Zamasu93** [comment]
I have some models for maglev trains which might a nice idea to add to traincraft because they would also tie to the addition of new track types and a faster train to use.

**Comment by EternalBlueFlame** [comment]
@Zamasu93 maglev and monorail trains and rollingstock are planned for the rewrite. unfortunately we can't do it in the port due to the shoddy track.

**Comment by Zamasu93** [comment]
I see, nice to know that they are planned.

**Comment by Bendragon1337** [comment]
How can I obtain a template of the current liveries and submit new versions of the liveries?

**Comment by Zamasu93** [comment]
I have them saved on my laptop and I could give them. They are a file that can be looked at because I am terrible at coding, so you can make a template based on the models.

**Comment by NitroxydeX** [comment]
@Bendragon1337 just download it here from the GitHub. We don't have a template for that just use one of our liveries and try it out. There are several programs that will assist you with the java-model too. I'm sure @Zamasu93 will help you

**Comment by balaji-venkatesh** [comment]
# MP40PH-3C
![image](https://cloud.githubusercontent.com/assets/17027186/26105876/e3ffac7c-3a11-11e7-83ee-0a8507c8394a.png)
It is a passenger train most commonly known as the GO Train.
It's also used in Sounder Transit, and other transit companies.
The MP40PH-3C is diesel; it's very similar to the IC4

# Info
**Top Speed**
150 km/h
**Horsepower**
4000, suggested 400
**Gauge**                    
4 ft 8 1⁄2 in (1,435 mm) standard gauge
**Wheel diameter**	
3 ft 4 in (1,016 mm)
**Minimum curve**	
248 ft (75.59 m)
**Wheelbase**	        
9 ft 0 in (2.74 m) (original)
9 ft 4 in (2.84 m) (new)
**Length**	                
68 ft (21 m)
**Width**	                 
0 ft 7.5 in (3.239 m)
**Height**	                
15 ft 6 in (4.72 m)
**Loco weight**        	
285,000 to 295,000 lb (129,000 to 134,000 kg)

# Images
Loco:
![image](https://cloud.githubusercontent.com/assets/17027186/26106221/30c78646-3a13-11e7-9198-a732e94099c8.png)

Cabs:
![image](https://cloud.githubusercontent.com/assets/17027186/26106272/5c4036ba-3a13-11e7-8615-ab336fd7e4c4.png)

Another Texture:
![image](https://cloud.githubusercontent.com/assets/17027186/26106327/8f0a9fcc-3a13-11e7-87a2-0ffde8856712.png)

Another Texture:
![image](https://cloud.githubusercontent.com/assets/17027186/26106390/c6dc7a1a-3a13-11e7-89f0-d7f283b1b6d7.png)

Another Texture:
![image](https://cloud.githubusercontent.com/assets/17027186/26106417/d4476552-3a13-11e7-8cf8-11da68680a80.png)

Another Texture:
![image](https://cloud.githubusercontent.com/assets/17027186/26106442/ebb0592e-3a13-11e7-9f10-f016183f33e3.png)

Another Texture:
![image](https://cloud.githubusercontent.com/assets/17027186/26106541/2ee29d2e-3a14-11e7-9df7-0e277853dc85.png)

Internals:
![image](https://cloud.githubusercontent.com/assets/17027186/26106844/46be14e0-3a15-11e7-8b5d-e9b1ed2f70fa.png)

**Comment by balaji-venkatesh** [comment]
# Bombardier Flexity Freedom
The Flexity Freedom is a low-floor, articulated light rail vehicle developed by Bombardier Transportation for the North American market. (It's Canadian)
![image](https://cloud.githubusercontent.com/assets/17027186/26107640/fbd531cc-3a17-11e7-88bd-7f0bd1631fb4.png)


# Info:
**Max Speed**
80 km/h
**Horsepower**
Low, something like 150.
**Car length**	
30.8 m (101 ft 1 in)
**Width**	
2.65 m (8 ft 8 in)
**Height**	
3.6 m (11 ft 10 in)
**Fuel**
Electricity
**Current collection method**
Pantograph
**Minimum turning radius**
25m

# Specific Stuff
It has three types of cabs, one with a door, one without a door, and one with a door and panto-graph.
The loco's themselves hold passengers.
The train comes in 3 and 5 module units, as seen below.
Those units themselves are linked together into longer trains.
There are very many colour schemes.

# Images
Side:
![image](https://cloud.githubusercontent.com/assets/17027186/26107979/1cabafce-3a19-11e7-9290-3525ec7cc3b5.png)
Front:
![image](https://cloud.githubusercontent.com/assets/17027186/26108147/a0cd2756-3a19-11e7-9d58-83b16190d0c4.png)
Driver's Interior:
![image](https://cloud.githubusercontent.com/assets/17027186/26108371/4d6a6e2e-3a1a-11e7-8b0a-67c09867154f.png)
Passenger Interior:
![image](https://cloud.githubusercontent.com/assets/17027186/26108430/80414d90-3a1a-11e7-9f64-f7b6ab3311e6.png)
![image](https://cloud.githubusercontent.com/assets/17027186/26108441/8697103a-3a1a-11e7-9662-aa91c1a4c0b5.png)

**Comment by trainsa** [comment]
# German Class 103

The BR 103 is an german high speed electro locomotive.
It was build by Henschel and Siemens.
## Technical Specifications

### Max Speed
200 km/h
### Horse Power
roundabout 8800
### length
19,m (longer version 20.2m)

## Pictures
Paintworks:
http://www.bahnbilder.de/bilder/db-br-103-233-3-am-10697.jpg
http://www.bahnbilder.de/bilder/db-br-103-148-3-am-5918.jpg
http://www.bahnbilder.de/1024/die-103-113-7-bei-lokparade-394172.jpg

Drivers Place:
https://upload.wikimedia.org/wikipedia/commons/c/cb/F%C3%BChrerstand_Baureihe_103.jpg

## Coaches
Rheingold paintwork: http://www.reisezugwagen.eu/wp-content/uploads/Avmz-111-61-80-19-95-035-0-Essen-18.6.94-1011.jpg
Older InterCity paintwork: http://www.bahnbilder.de/1200/friseurwagen-avmz-618084-95031-9-am-1151991-883831.jpg
Actuall InterCity paintwork: http://deutsche-reisezugwagen.de/uploads/images/100/109.2_Avmz__61_80_19-95_256-2__Hannover_Hbf__26.05.2015.jpg
Alex Train paintwork: http://www.regensburger-busse.de/forum/alex_avmz04.JPG

**Comment by EternalBlueFlame** [comment]
requested by a user via Discord DM:
Model : GM-EMD GT18LA-2 
Builders : GM-EMD

Production year :2013
Country of Origin : USA
Engine : EMD 8-645E3C
Cylinders : V8
Power output : 1,500 hp (1.12 MW)
Engine type : 2-stroke diesel
Aspiration : Roots-type supercharger
Fuel Capacity : 3000 Ltr
Transmission :DC traction motor
Top speed : 107 km/h (depends on load)
Weight: Approx. 140 Metric Ton
Tractive effort :438 kn (98,000 lbf)
Wheel : A1A-A1A
Locomotive brake : Straight air
Train brakes : Straight air

https://cdn.discordapp.com/attachments/308459012436000768/316454452456325120/14601029_876029562531815_4883788124148747273_n.jpg
https://cdn.discordapp.com/attachments/308459012436000768/316452575442501642/unknown.png
https://cdn.discordapp.com/attachments/308459012436000768/316454822910099456/14650321_876028222531949_4581506679013898653_n.jpg
https://images.discordapp.net/.eJwNzNsNhCAQAMBeKAB23RfajCFq0IuKEfy6XO_nFDBf99y7G9za2lWHEOatTuWefW3lTnnxuZS8L-naqp_KEVJraVqP5Ww1EESWHrBjUgAwjYFQWQQ0clRWk94CMkJnnY5RgJBEBYEYZTQhE2Ozd0F7D6Tx9J8ru98fkggp4g.OlB1-hnFCrwOvbe1we-0YxsCuzo?width=901&height=676

**Comment by Anoobarmy** [comment]
# AVE series 103

From renfe the leading transport-by-rails company here in spain here are the especifications

# Specifications

max speed 350 km/h

 550 kW electric motor

acceleration from 0 to 320 kmh: 380 s.

# dimensions

height: 3.890 mm

locomotive lenght: 25.675 mm

car lenght: 24.775 mm

aluminium casing

# Photographs

Exterior of locomotive
![image](https://user-images.githubusercontent.com/29972393/27947519-74cbd2dc-62f6-11e7-9bb1-f3989fea0bf6.png)

Cafeteria:
![image](https://user-images.githubusercontent.com/29972393/27947562-a4527b00-62f6-11e7-9aa9-2cf02db64a4c.png)
High  class wagon:
![image](https://user-images.githubusercontent.com/29972393/27947587-bd8dcef8-62f6-11e7-84df-b0d27ef0b80f.png)

low class wagon:
![image](https://user-images.githubusercontent.com/29972393/27947595-c9939dcc-62f6-11e7-8da2-9c0b61a66da0.png)

it should be like the hi speed train but woth low comsumption and hi speed,prepared for long trains (regular train lenght:200m)

i do not have models sorry.

# My thoughts about how it should be

it should have low fuel contsumption/high eficiency
Becouse these trains have no stops and travel long distances EX: from spain to madrid Daily

its brakes shouldnt be too good though and need large stations and very long curves to not derail becouse its hi speeds

source: (in spanish) http://www.renfe.com/viajeros/nuestros_trenes/aves103_ficha.html

**Comment by EternalBlueFlame** [comment]
@Anoobarmy, I noticed you posted your request here, and in 3 separate issues.
There's this thing called "manners" you should check it out, google it or something.
Just so we're clear, spamming your request doesn't make us add it faster. If anything we'll likely be less likely to add your request of [Generic Inter-city train that looks like a re-textured IC4].

**Comment by Anoobarmy** [comment]
@EternalBlueFlame  Hi im sorry for that i wasnt spamming the thing is that im new to github and i was weirded out becouse my comment wasnt appearing on here, i was making new threads i didnt notice that the people were commenting and i thought it was the button "New issue" im really sorry its my fault.

PS: I dont travel too much out of spain so i did not know that renfe or the same kind of trains were in other places, Its also my fault to be so generic sorry
Also the new inter city train from the last version, i think it should be electric becouse it looks like it,
(Im imagining me on a hall of shame kind of thing)

**Comment by EternalBlueFlame** [comment]
@Anoobarmy I see. My mistake then, sorry. I'll retract my previous comment, if that's the case, it's out of line. I won't delete it though, no point in hiding my own failures.
Thanks for the clarification.

**Comment by Anoobarmy** [comment]
@EternalBlueFlame Dont fell bad you are a nice guy, but lets stop this conversation,it would be endless.
(delete this comment after reading it, it does not velong here. if you want to answer email me:ale.rod.duran@gmail.com)
PS: Lol fast answer
PPS:I googled manners L.O.L

**Comment by reboplic** [comment]
A BR 41 Oil would be nice.

http://www.eisenbahnjahre.de/wp-content/uploads/2013/11/Roco-BR41-oel-1968.jpg

or a V188

http://www.e94114.de/V188/V188001rot_Bruenning.jpg

with 1600 PS

The Fuel are heating oil

**Comment by Jackboy317** [comment]
What about a PRR K4s 4-6-2 or (even more awesome) PRR T1 4-4-4-4 duplex
Or a New York central class J1e Hudson 4-6-4 ( try to make the Hudson a reskin of the c62)

**Comment by Jackboy317** [comment]
Here's the links to all: (also make sure the Hudson is unstreamlined) 
Hudson https://en.m.wikipedia.org/wiki/New_York_Central_Hudson

K4s (PRR) https://en.m.wikipedia.org/wiki/Pennsylvania_Railroad_1361

PRR T1 
https://en.m.wikipedia.org/wiki/Pennsylvania_Railroad_class_T1

**Comment by Aldaaaaaa** [comment]
My wish is Italian highspeed trains.

Frecciarossa 1000 ETR400 from Trenitalia is the new Highspeed train. The train has a max speed of 360km/h, Pleas Scrooling under:  https://www.transportfever.net/filebase/index.php/Entry/460-ETR-1000-Frecciarossa-v2-0-Build-5068/

Frecciarossa ETR500: Is tje older highspeed train. The max speed is 300km/h : https://en.m.wikipedia.org/wiki/Frecciarossa

And the last is a Italian locomotive the E656: https://en.m.wikipedia.org/wiki/FS_Class_E.656

This are my biggest wish, the E656 is a additive. The Frecciarossa must by! 

Thanks

**Comment by Valaktik** [comment]
Wait ... What "user input" are you waiting about #294 ? I though it was ok xD
###### _Added after 10 mins :_
Oooooh ... The shapeboxes issue ! Isn't it ? Okey...
###### _Added after 50 mins :_
By the way, I can't type on #294 ...
Does anyone know how to make lights with SMP ToolBox ? Like on the BR E69.

**Comment by Valaktik** [comment]
Oooooh ... The shapeboxes issue ! Okey...

**Comment by EternalBlueFlame** [comment]
@Valaktik not sure why you cant type in the issue..? i can close it and you can open a new one if you want, it's not a big deal.

I've been working on a guide on my blog if that is of any help:
http://eternal-b-flame.blogspot.com/p/modelling.html

As for the "lights" it's a contrast effect in the texture, take a look at the texture from the high speed train, it works as a good example.

**Comment by NitroxydeX** [comment]
@Valaktik @EternalBlueFlame he can't type in there because I locked that topic to collaborators. So his ticket isn't spammed with "nice train" "good work man" and so on.

**Comment by Valaktik** [comment]
@NitroxydeX , makes sense.

```javascript
Minecraft.getMinecraft().entityRenderer.disableLightmap(1D);
box28.render(f5);
box29.render(f5);
Minecraft.getMinecraft().entityRenderer.enableLightmap(1D);
```
@EternalBlueFlame Is this how headlights works ? It's in java codes.

**Comment by NitroxydeX** [comment]
@Valaktik yes that's how Lightmaps are working. Between the disable and enable statement the boxes for the Lightmap are defined and magical Minecraft is doing the rest.

**Comment by Valaktik** [comment]
@NitroxydeX Yippee !!! 😃

**Comment by EternalBlueFlame** [comment]
Well the thing is that minecraft uses some hacky non OpenGL based lightmapping, like for night/day stuff. By disabling that, the light appears brighter because it doesn't get tinted by the hacky night/day lightmapping.
Although I have absolutley no idea that was even an option, so that's an awesome find.

**Comment by Admiral61** [comment]
I've always had a slight problem with TrainCraft that would make the game a lot better in my experience. And that problem is steam locomotives from the US. There are eight steamers thus far in the mod, three of them are from the US, but they are all small and from the wild west era (4-4-0, Mogul, Small Steamer).
I would like to suggest a newer, larger locomotive. My suggestion is Nickel Plate Road 587, or a locomotive like 587.
https://i.pinimg.com/originals/85/c0/30/85c0308a2b4419c9c5e338d0c2c4bb12.jpg
No. 587 is a USRA, Baldwin Locomotive Works built H6-o class 2-8-2 Light Mikado originally in service with the Lake Erie and Western Railroad as No. 5541, but when the LE&W was bought by the Nickel Plate Road, her number was changed to 587.

Locomotive stats:
Type: H6-o class 2-8-2 Light Mikado
Weight (fully loaded): 300,900 lbs (136486 kg)
Engine Length Coupler to Coupler: 90 feet, 6 inches (27.6 meters)
Height: 14 feet, 11 inches (4.5 meters)
Horsepower (guesstimating based on what research I could gather): 2500-3500 HP
Top Speed (guesstimating based on what research I could gather): 70-80 MPH (112.6-128.7 KMH)

Tender stats:
Type: 22RA Tender
Coal Capacity: 20 tons of Coal (18,143.7 kg)
Water Capacity: 22,000 gallons (83,280 liters)

Thank you for reading this, if you are still accepting ideas, and sorry for any inconvenience that I may have caused. I gathered what information I could to help but sadly it's not much. 

If you need more visuals of the locomotive, the link here will provide plenty of images.
 http://www.railpictures.net/showphotos.php?road_number=NKP%20587

**Comment by minningguyteen2** [comment]
how about this one:
![image](https://user-images.githubusercontent.com/28766861/33577484-0a771174-d93b-11e7-8164-c2fa88e7bed6.png)

https://en.wikipedia.org/wiki/British_Rail_Class_43_(HST)
type: diesel
top speed: 125mph (201kph

coaches it uses

British railmk3
https://en.wikipedia.org/wiki/British_Rail_Mark_3
![image](https://user-images.githubusercontent.com/28766861/33577539-3f346df8-d93b-11e7-84ac-7c00b22e2b37.png)

skins they will use: British rail, Intercity swallow, First great western, GWR and Virgin east coast

specs:
Configuration:
​
 • UIC	Bo'Bo'
 • Commonwealth	Bo-Bo
Gauge	1,435 mm (4 ft 8 1⁄2 in) standard gauge
Bogies	BP10
Wheel diameter	3 ft 4 in (1.016 m)
Length	17.79 metres (58.4 ft)
Width	2.74 metres (9.0 ft)
Loco weight	70.25 tonnes (69.14 long tons; 77.44 short tons)
Fuel capacity	990 imp gal (4,500 l; 1,190 US gal)
Prime mover	Paxman 12VP185 or MTU 16V4000 R41R
Originally Paxman Valenta 12RP200L
Alternator	VP185: Brush Traction BA1001B
MTU: Brush Traction BA1001C
Traction motors	GEC G417AZ (43124-43152)
Brush Traction TMH68-46
Both frame mounted, four off.
Transmission	Engine driven alternator and rectifier supplying DC to traction motors
MU working	Within class only
Train heating	Electric Train Heat
Train brakes	Air

some helpful images:
![image](https://user-images.githubusercontent.com/28766861/44302521-5e3be800-a322-11e8-93cf-9475ea9bfc40.png)

![image](https://user-images.githubusercontent.com/28766861/44302526-89263c00-a322-11e8-9b2c-47ad31eac1c2.png)

![image](https://user-images.githubusercontent.com/28766861/44302528-9b07df00-a322-11e8-9209-60909f0571c6.png)

**Comment by reboplic** [comment]
How about this one?
UP-Klasse 4000 „Big Boy“

**Info**

Hersteller: | ALCO
-- | --
Baujahr(e): | 1941, 1944
Achsformel: | (2’D)D2’ h4
Spurweite: | 1435 mm (4 ft 8½ in)
Länge über Kupplung: | 40.500 mm
Höhe: | 4940 mm
Dienstmasse: | 350,2 t
Dienstmasse mit Tender: | 548,3 t
Höchstgeschwindigkeit: | 130 km/h
Anfahrzugkraft: | 602 kN
Kesselüberdruck: | 21,1 bar
Rostfläche: | 13,97 m²
Überhitzerfläche: | 229 m²
Verdampfungsheizfläche: | 547 m²
Tender: | Centipede
Wasservorrat: | 94,6 m³
Brennstoffvorrat: | 25,4 t Kohle

[https://en.wikipedia.org/wiki/Union_Pacific_Big_Boy](url)


![up4023_big_boy](https://user-images.githubusercontent.com/30384930/33399710-e89caa5e-d553-11e7-9746-10d5ba4161f7.jpg)
![maxresdefault](https://user-images.githubusercontent.com/30384930/33399711-e8d0e9d6-d553-11e7-9cab-fd4120d5c231.jpg)
![hr2358_1052034_qty1_2](https://user-images.githubusercontent.com/30384930/33399712-e8eeeb0c-d553-11e7-8944-a85a39d44f55.jpg)

**Comment by TheSpareAce** [comment]
The UP Big Boy is a beautiful train. 

Here's an electric train I want. The PRR GG1.  It is a streamlined electric locomotive capable of hitting 160km/h, with 291kN and around 4,600 HP, weighing 215 tonnes. It's default paint scheme should be black, with the option of blue, red and white.

![image](https://user-images.githubusercontent.com/25802995/33965615-6845fec8-e011-11e7-855b-f9fcf7f13b13.png)

![](https://media-cdn.tripadvisor.com/media/photo-s/0e/aa/47/fb/prr-gg1.jpg)

![](http://members.localnet.com/~docsteve/railroad/gifs/PRR_GG1.gif)

**Comment by Zamasu93** [comment]
Another note I may add is the diesel engines can use fuel from other mods which I like, but it would be nice if the b-units could also since their engines can

**Comment by NitroxydeX** [comment]
@Zamasu93 that's a good hint. Thanks for the headsup!

**Comment by Zamasu93** [comment]
I noticed that when I try the new trains and I played around with the b-units, that is where I noticed that they only worked with the diesel fuel and sadly did not even work with the refined fuel from this mod while the other diesel engines did. I like this mod a lot so every time there is an update I see what is new and how they work

**Comment by EternalBlueFlame** [comment]
That is a heavily requested feature, given the limitations of it being a glorified tanker car I wasn't exactly sure how to go about using a fluid filter similar to the diesel trains, I was putting it off till I thought of a way.
Good news is, I _finally_ got it working for 015, while I was in there I found a GUI bug with tankers and an inventory bug with the B-units too, so..... bonus?
Currently it's only on the DD35, but i'll get it on the rest when I have time.

Had to basically replace the entire fluid management system for the thing.

**Comment by Zamasu93** [comment]
no problem, I like to be helpful when I can be. Since I have been a bit helpful lately in another mod by making some textures so playing mods also helps a lot with noticing problems especially when it is played by others.

**Comment by melindaendv** [comment]
Add this to the USA Propaganda cars - https://oneway2day.files.wordpress.com/2016/06/trump-build-that-wall.jpg

or

https://scholarsandrogues.files.wordpress.com/2016/12/flat800x800075f.jpg

**Comment by EternalBlueFlame** [comment]
I already thought about doing that along with a few others, but it's too relevant and creates a lot of butthurt in the community.

**Comment by minningguyteen2** [comment]
How about this one (Yes i know it tilts but we don't need the tilting mechanism just yet.):

![image](https://user-images.githubusercontent.com/28766861/34528932-5ef62c66-f0a2-11e7-9221-d4ceafb29dde.png)

Train length | 482 ft 0 1⁄2 in (146.93 m)[2]
-- | --
3.5 m (11 ft 5 3⁄4 in)
155 mph (249 km/h) (Design)125 mph (201 km/h) (service)
434 tonnes (427 long tons; 478 short tons)
4 × Asea LJMA410F[2]
25 kV 50 Hz AC Overhead
Pantograph
2'2'2'2'2'2'2'+Bo'Bo'+Bo'Bo'+2'2'2'2'2'2'2'
BP17a, BT11 and BT12[2][4]
Air/hydrokinetic[6]Electronic control[4]
Automatic tightlock (M)Automatic drophead buckeye (DTS)Bar (other)[4]
Within class only
1,435 mm (4 ft 8 1⁄2 in) standard gauge

![image](https://user-images.githubusercontent.com/28766861/34528964-73e70fb4-f0a2-11e7-89da-133eb31dd85e.png)
![image](https://user-images.githubusercontent.com/28766861/34528971-79a21822-f0a2-11e7-8b74-b6f8b92cbbc9.png)

**Comment by LeadKronos** [comment]
I would like to see the addition of the C&O 2-6-6-6 Allegheny locomotive
![image](https://user-images.githubusercontent.com/39572016/40485023-63b3b046-5f2b-11e8-87cd-d84d0be8874f.png)
this is the information I could find 
Railroad of Record: Chesapeake & Ohio Railway

Type of Locomotive: 2-6-6-6 Allegheny (simple articulated)
Class: C&O Class H-8
Builder Number: 7824
Manufactured by: Lima Locomotive Works
Date Built: December 1941
Locomotive Weight: 389 tons
Driver Diameter: 67 inches
Cylinders: 22 ½ x 33 inches (4)
Tractive Effort: 110,200 lbs
Max Speed: unknown but one source notes that it can go atleast 45mph
Horsepower Rating: 7,500

**Comment by byarchitect** [comment]
> resolution

Thank you so much. I'm finding drawing for modelling. Do you have cad drawing or bigger image ? Thank you!

**Comment by Admiral61** [comment]
I couldn't find any CAD drawings but I hope this will work. 
https://kltrainz.com/wp-content/uploads/2017/09/587-1.jpg

    On Saturday, March 23, 2019, 12:20:15 AM PDT, byarchitect <notifications@github.com> wrote:  
 
 

resolution


Thank you so much. I'm finding drawing for modelling. Do you have cad drawing or bigger image ? Thank you!

**Comment by byarchitect** [comment]
> I couldn't find any CAD drawings but I hope this will work.  https://kltrainz.com/wp-content/uploads/2017/09/587-1.jpg On Saturday, March 23, 2019, 12:20:15 AM PDT, byarchitect <notifications@github.com> wrote: resolution Thank you so much. I'm finding drawing for modelling. Do you have cad drawing or bigger image ? Thank you! — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or mute the thread.

Thank you but it's not model i'm finding.
I'm looking for  D19E-961 but can't find drawing.
I'm living in Vietnam, easy for take a photo but difficult for find technical information lol.

**Comment by minningguyteen2** [comment]
heres a third (i actually have loads but don't want to barrage you lot with lol.)

TGV TMST (Class 373 eurostar E300.)

maximum speed: 300kph (186 mph)

year built: 1992- 1996

type: electric

seating capacity:
750 seats (Three Capitals - Classic)
758 seats (Three Capitals - Refurbished)
558 seats (North of London)


Car body construction | Steel
-- | --
387 m (1,269 ft 8 in)
18.7 m (61 ft 4 in) (middle)[3]22.15 m (72 ft 8 in) (driving)21.84 m (71 ft 8 in) (powered middle)
2.81 m (9 ft 3 in)
300 km/h (186 mph) (Service)334.7 km/h (208.0 mph) (Record)
752 t (740 long tons; 829 short tons) (Three Capitals, empty)
815 t (802 long tons; 898 short tons) (Three Capitals, loaded)665 t (654 long tons; 733 short tons) (North of London)
12.2 MW (16,400 hp) (25 kV)5.7 MW (7,600 hp) (3000 V)[4]3.4 MW (4,600 hp) (750 V)[4]Tractive Effort:410 kN (92,000 lbf) Starting @ 25 kV 350 kN (79,000 lbf) Starting @ 1.5 kV & 750 V220 kN (49,000 lbf) Continuous @200 km/h (124 mph) & 25 kV[4]
Overhead lines25 kV 50 Hz AC 3000 V DC, 1500 V DC Third rail 750 V DC (no longer used)
PantographContact shoe (removed)
Bo'Bo'+Bo'2'2'2'2'2'2'2'2'2'+2'2'2'2'2'2'2'2'2'Bo'+Bo'Bo'Bo'Bo'+Bo'2'2'2'2'2'2'2'+2'2'2'2'2'2'2'Bo'+Bo'Bo'
Scharfenberg
1,435 mm (4 ft 8 1⁄2 in) standard gauge

liverys:
 bone meal for the original (old livery.) no logos
 rose red dye for the white rose
lapis azuli for the new livery
grey for the SNCF livery and Dandelion Yellow dye for the old livery and old logos.

heres some pictures:
![class 373 coaches](https://user-images.githubusercontent.com/28766861/57489342-a387b780-72ad-11e9-8bab-818ee070cc81.jpg)
![class 373 white rose](https://user-images.githubusercontent.com/28766861/57489343-a387b780-72ad-11e9-8381-9ce48b5726d7.jpg)
![class 373](https://user-images.githubusercontent.com/28766861/57489345-a4204e00-72ad-11e9-95d8-bc6a53714f03.jpg)
![new livery 2](https://user-images.githubusercontent.com/28766861/57489364-aedae300-72ad-11e9-8c9a-71145f2c0223.jpg)

**Comment by NLSentryman** [comment]
_Love the new models! The updated Rheingold and BR 103 are especially well done! Good job!_

How about some classic, stylish steam engines?
I'd love to see the German Bavarian S3/6. Here's some pictures and specs:
*specs and drawing are from a different variation of engine but very close to the blue one i mentioned.

![roco_ro61471-1](https://user-images.githubusercontent.com/57464634/68346098-8a907f80-00f3-11ea-8051-2c6141ffc078.jpg)
![stoomlocomotief-S3-6-Salonwagen-Koenig-Luitpold-K-Bay-Sts-B-AC-Digital-Sound-Roco-61473_b_0](https://user-images.githubusercontent.com/57464634/68346100-8a907f80-00f3-11ea-8412-bc4db29a3781.jpg)
![Roco-S-3-6-61471-61472-61473-Front-Oben](https://user-images.githubusercontent.com/57464634/68346099-8a907f80-00f3-11ea-9bcc-cc2bab878871.jpg)
![br18-4](https://user-images.githubusercontent.com/57464634/68346096-8a907f80-00f3-11ea-8f90-15042c0f4551.jpg)
![S36](https://user-images.githubusercontent.com/57464634/68346202-cd525780-00f3-11ea-96a9-ca7a71fdf1c2.png)

_Now if only i had some modeling skills grrr..._

EDIT: Some more specific whistles / horns would be nice. The BR103 sounds like a BR185. I could help with making sound files if neccesary. Allready made a few to replace some sounds.

**Comment by jatiwarnaerep** [comment]
hello, i've noticed there's no indonesian trains yet, because mostly were americans trains. so you might wanna add these two types of locomotives in traincraft (and maybe some indonesian rolling stocks) with the short locomotive style to fit on curves!
GE CM20EMP, also known as CC206, maximum speed of 120 KM/H
![Logo-KAI-di-Loko-CC206](https://user-images.githubusercontent.com/91058136/133982899-cafb4b82-5c4e-47e8-8483-4adcbb90745d.jpg) 
 GE U20C, also known as CC203, the same maximum speed of 120 KM/H
![50415034862_72cb554305_b](https://user-images.githubusercontent.com/91058136/133983091-53df8145-12b5-4d00-8c54-1e07606c1bbc.jpg)
diagrams of both locomotives can be seen here:
![image](https://user-images.githubusercontent.com/91058136/133983306-50a06111-d07c-432f-833e-2f5d59cb05b8.png)
![image](https://user-images.githubusercontent.com/91058136/133983443-55066fb5-8d2c-4daf-92e2-defab20c6f76.png)
coaches (optional):
![image](https://user-images.githubusercontent.com/91058136/133983778-b8671051-4bb5-445b-9462-f7a84d9756ec.png)
![image](https://user-images.githubusercontent.com/91058136/133983722-7ec9d7a1-a4ac-49c6-9a48-91bd13fd2df6.png)
![image](https://user-images.githubusercontent.com/91058136/133983846-7f136a57-239b-4d14-96b7-90a9e7440a48.png)


that's it, thank you very much, have a good day!

its okay if you brush this off, i can totally understand that :D

**Comment by byarchitect** [comment]
I can believe you still develop this website.
Thank you so much!
Have a niceday!

Vào Th 2, 20 thg 9, 2021 vào lúc 16:51 jatiwarnaerep <
***@***.***> đã viết:
